frontend/apps/reader/modules/readersearch.lua

```diff
@@ -271,6 +278,7 @@ function ReaderSearch:onShowFulltextSearchInput()
 
     UIManager:show(self.input_dialog)
     self.input_dialog:onShowKeyboard()
+    return true
```

Comment:
@poire-z this line stops the shortcut press from bleeding into the search field, are you happy with it?

Reply:
It feels it is indeed the right solution: i guess the key event handler will get the return value from that handler, and if true, will stop propagating the key.

Reply:
As far as I understand, this line doesn't affect the list of arguments passed to the event handler by `InputContainer:onKeyPress`.
The key is always passed as the last (or the only) arg (via pack/unpack in Eventlistener)
https://github.com/koreader/koreader/blob/5affb68233fda994150df1da3d0b573402f20842/frontend/ui/widget/container/inputcontainer.lua#L249
https://github.com/koreader/koreader/blob/5affb68233fda994150df1da3d0b573402f20842/frontend/ui/widget/eventlistener.lua#L37

Reply:
May be it's there on lines 101 and then 83-85 that the magical stop happens:
https://github.com/koreader/koreader/blob/5affb68233fda994150df1da3d0b573402f20842/frontend/ui/widget/container/widgetcontainer.lua#L80-L103

Reply:
Yeah, true eats it as handled.

Reply:
Input field is empty because `empty_input` is truthy 
https://github.com/koreader/koreader/blob/09a35836401a20f94ff11a6b895e8bc932628224/frontend/apps/reader/modules/readersearch.lua#L224

Reply:
But this solution kills the idea of using previous search strings.

Reply:
> true eats it as handled

Sure, but the question is about the key "bleeding" (as the author calls it).

Reply:
> this line stops the shortcut press from bleeding into the search field

Btw, how can you say that, I thought you didn't get that issue on Kindle ?

It still doesn't solve anything for me (after installing this PR): with or without the `return true`, same behaviour on the emulator that I described ("s" bleeding on first launch after startup, not on next launches).
And I still can't figure out why :)

Reply:
```suggestion
```

As per https://github.com/koreader/koreader/pull/12162#issuecomment-2243812140 please remove this.

Reply:
> Btw, how can you say that, I thought you didn't get that issue on Kindle ?

you must have missed my update to that, essentially i spoke too soon. It removed the last search but not the carry over key input from the shortcut key press. I mean we had it with “Alphanumeric” so… The `return true` definitely solves it now (on kindle)